cou/apps/auxiliary.py

```diff
@@ -93,3 +94,49 @@ def channel_codename(self) -> OpenStackRelease:
                 f"for channel: '{self.channel}'"
             )
         )
+
+
+@AppFactory.register_application(["rabbitmq-server"])
+class RabbitMQServer(OpenStackAuxiliaryApplication):
+    """RabbitMQ application.
+
+    RabbitMQ had to wait until the entire model will be idle after the upgrade to ensure other
+    upgrade can continue.
+    """
+
+    def post_upgrade_plan(self, target: OpenStackRelease) -> list[Optional[UpgradeStep]]:
+        """Post Upgrade planning.
+
+        Wait until the entire model reaches the idle state and then check the target workload.
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :return: Plan that will add pre upgrade as sub steps.
+        :rtype: list[Optional[UpgradeStep]]
+        """
+        return [
+            self._get_wait_step(300, wait_for_itself=False),
+            self._get_reached_expected_target_plan(target),
+        ]
+
+
+@AppFactory.register_application(["keystone"])
+class Keystone(OpenStackApplication):
```

Comment:
Doesn't make sense having Keystone on the auxiliary module

Reply:
I do not understand what is auxiliary and why we have ceph.py where we are using OpenStackAuxiliaryApplication. Do you have any suggestion? To my we should have base.py, auxiliary.py, principal.py and subordinate.py. What do you say? Maybe some other suggestion?

Reply:
Auxiliary charms are those that makes part of OpenStack project, but are not directly connected with the o7k releases like rabbitmq, ceph, mysql and etc. On those charms we have 1:N mapping where one workload version can match with multiple OpenStack releases. On the other hand, OpenStack applications have a 1:1 mapping where a workload version matches with just one OpenStack release.

#95 tries to destroy the need of new classes for subordinates and I think it's more maintainable the project if we have fewer classes.